### PR TITLE
feat(react-color-picker): added ColorSlider

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
This is simple ColorSlider based on a Slider component.
What was done:
- Hue channel is supported by default
- Change of thumb color based on selected channel
- Vertical and horizontal slider
- Selection of value

image

Features, which will be supported in following PRs:
- Support of other channels and custom ones
- Context and sharing data with the root component
- Focus style for thumb

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts [hunk 1]
```diff
@@ -1,17 +1,71 @@
...
+  thumb: NonNullable<Slot<'div'>>;
+  input: NonNullable<Slot<'input'>> & {
+    /**
+     * Orient is a non standard attribute that allows for vertical orientation in Firefox. It is set internally
+     * when `vertical` is set to true.
+     * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range#non_standard_attributes
+     * Webkit/Chromium support for vertical inputs is provided via -webkit-appearance css property
+     */
+    orient?: 'horizontal' | 'vertical';
+  };
```
Comment (ValentinaKozlova) [inline packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts:22]: This is part of existing Slider

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const _onChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(ev => {
+    const newValue = Number(ev.target.value);
+    setCurrentValue(clamp(newValue, min, max));
+
+    if (inputOnChange && inputOnChange !== (onChange as unknown as React.ChangeEventHandler<HTMLInputElement>)) {
+      inputOnChange(ev);
+    } else if (onChange) {
+      onChange(ev, { type: 'change', event: ev, value: newValue });
+    }
+  });
```
Comment (ValentinaKozlova) [inline packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx:39]: This is part of existing Slider

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  const rootVariables = {
+    [sliderDirectionVar]: state.vertical ? '180deg' : dir === 'ltr' ? '-90deg' : '90deg',
+    [sliderProgressVar]: `${valuePercent}%`,
+    [thumbColorVar]: `hsl(${clampedValue}, 100%, 50%)`,
```
Comment (ValentinaKozlova) [inline packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx:44]: Currently this color is based on hue, but when more channels added, it'll be changed.

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderStyles.styles.ts [hunk 4]
```diff
@@ -1,22 +1,192 @@
...
+    // 3x3 grid with the rail and thumb in the center cell [2,2] and the hidden input stretching across all cells
+    gridTemplateRows: `1fr var(${thumbSizeVar}) 1fr`,
+    gridTemplateColumns: `1fr calc(100% - var(${thumbSizeVar})) 1fr`,
+  },
+
+  vertical: {
+    minHeight: '280px',
+    // 3x3 grid with the rail and thumb in the center cell [2,2] and the hidden input stretching across all cells
+    gridTemplateRows: `1fr calc(100% - var(${thumbSizeVar})) 1fr`,
+    gridTemplateColumns: `1fr var(${thumbSizeVar}) 1fr`,
```
Comment (ValentinaKozlova) [inline packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderStyles.styles.ts:68]: Styles of existing Slider

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts [hunk 5]
```diff
@@ -1,17 +1,71 @@
...
+export type SliderOnChangeData = EventData<'change', React.ChangeEvent<HTMLInputElement>> & {
+  value: number;
+};
 
 export type ColorSliderSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'div'>>;
+  rail: NonNullable<Slot<'div'>>;
+  thumb: NonNullable<Slot<'div'>>;
+  input: NonNullable<Slot<'input'>> & {
```
Comment (marcosmoura) [inline packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts:14]: Is this a component prop/attribute? Should this be part of the Slots definition?

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
```
Comment (marcosmoura) [inline packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/useColorSliderState.tsx:1]: NIT: This file doesn't need to be .tsx and can simply be .ts instead.

File: packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts (hunk 5 above)
Comment (ValentinaKozlova) [reply packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts:14]: I asked Micah (as it was part of the Slider component). It's for internal use only, so I'll remove it

**Review by marcosmoura — APPROVED** [review]

**Pull request merged**